src/app/_hooks/useSetParams.js

```diff
@@ -0,0 +1,23 @@
+import { useRouter, useSearchParams } from 'next/navigation';
+
+export function useSetParam(param) {
+  const searchParams = useSearchParams();
+  const router = useRouter();
+  const newParams = new URLSearchParams(searchParams);
+  const addParam = value => {
+    if (searchParams.get(param) && param !== 'district') {
+      newParams.delete(param);
+    }
+    newParams.append(param, value);
+    router.push(`?${newParams.toString()}`);
+  };
+  const deleteParam = value => {
+    if (value) {
```

Comment:
🟠 do you need this if? won't `newParams.delete(param,value)` will work just fine?

Reply:
Again, districts filter is multiselect and we may want to remove only one district from this multiselect so we need to provide a value to be deleted, but when, for example, we click on "clear" button in some filter we want to delete all parameters